src/pyroboplan/planning/utils.py

```diff
@@ -54,3 +129,33 @@ def retrace_path(goal_node):
         current = current.parent
     path.reverse()
     return path
+
+
+def discretize_joint_space(model, step_size):
```

Comment:
I think this name doesn't immediately put out that this is a generator.

Wondering if this + `discretized_sample_generator` can be unified into one utility here of the same name?

The behavior of "returning random states after finishing the discrete grid" could simply be a boolean option that can be directly passed into this generator so you don't go through 2 layers of indirection, right?

Reply:
Also, should this have a unit test?